File: practice/data_structures/linked_lists/find_merge_point_of_two_lists/find_merge_point_of_two_lists.cpp
/*
   Find merge point of two linked lists
   Node is defined as
   struct Node
   {
       int data;
       Node* next;
   }
*/

int FindMergeNode(Node * i_headA, Node * i_headB)
{
    while (i_headA)
    {
        Node * node = i_headB;
        while (node)
        {
            if (i_headA == node)
            {
                return i_headA->data;
            }
            node = node->next;
        }
        i_headA = i_headA->next;
    }
    
    return -1;
}

File: practice/data_structures/linked_lists/find_merge_point_of_two_lists/find_merge_point_of_two_lists.py
"""
 Find the node at which both lists merge and return the data of that node.
 head could be None as well for empty list
 Node is defined as
 
 class Node(object):
 
   def __init__(self, data=None, next_node=None):
       self.data = data
       self.next = next_node

 
"""

def FindMergeNode(i_headA, i_headB):
    while i_headA:
        cp = i_headB
        while cp:
            if i_headA == cp:
                return i_headA.data
            cp = cp.next
        i_headA = i_headA.next

